pkg/config/config.go

```diff
@@ -396,6 +396,28 @@ func (c *config) loadFromFile(filename string, fsys fs.FS) error {
 	return c.loadFromReader(v, f)
 }
 
+// Used to improve config parsing errors into a more contextual and actionable
+// error message
+func wrapTomlError(err error) error {
```

Comment:
Can we implement this as a decode hook? By checking if the source type is a non-map value and the target type is `FunctionConfig`.

Reply:
I extracted the custom decode hooks into their own file so it's easier to find them and refactored to use a custom decode hook as you suggested here: https://github.com/supabase/cli/pull/3101/commits/c256ee3697efb3d38e67668059e12322a8f49c6e